src/HonzaBotner.Discord/ServiceCollectionExtensions.cs

```diff
@@ -15,18 +15,22 @@ public static class ServiceCollectionExtensions
         }
 
         public static IServiceCollection AddDiscordBot(this IServiceCollection serviceCollection,
-            Action<CommandsNextExtension> commandConfig, Action<ReactionListBuilder> reactionConfig)
+            Action<CommandsNextExtension> commandConfig, Action<ReactionListBuilder> reactionConfig, Action<IServiceCollection> other)
         {
             serviceCollection.AddHostedService<DiscordWorker>();
             serviceCollection.AddSingleton<IDiscordBot, DiscordBot>();
             serviceCollection.AddSingleton<DiscordWrapper>();
             serviceCollection.AddTransient<IGuildProvider, ConfigGuildProvider>();
             serviceCollection.AddTransient<ReactionHandler>();
+
             serviceCollection.AddSingleton(new CommandConfigurator(commandConfig));
+
             ReactionListBuilder builder = new(serviceCollection);
             reactionConfig(builder);
             serviceCollection.AddSingleton(builder.Build());
 
+            other(serviceCollection);
```

Comment:
If it is core service for Discord, register it inside method, without adding new lambda. Otherwise register it normally in Startup.cs

Reply:
Not sure how this works, so I temporarily did it this way. I would like to talk about it.